samcli/lib/sync/infra_sync_executor.py

```diff
@@ -101,6 +101,39 @@ def _boto_client(self, client_name: str, session: Session):
         """
         return get_boto_client_provider_from_session_with_config(session)(client_name)
 
+    def execute_infra_sync(self, first_sync: bool = False) -> bool:
+        """
+        Compares the local template with the deployed one, executes infra sync if different
+
+        Parameters
+        ----------
+        first_sync: bool
+            A flag that signals the inital run, only true when it's the first time running infra sync
+
+        Returns
+        -------
+        bool
+            Returns True if infra sync got executed
+            Returns False if infra sync got skipped
+        """
+        self._build_context.set_up()
+        self._build_context.run()
+        self._package_context.run()
+
+        # Will not combine the comparisons in order to save operation cost
+        if first_sync:
+            if self._auto_skip_infra_sync(
```

Comment:
Should we wrap this method with try/except so that any unexpected failure will be handled by running the InfraSync?

Reply:
auto_skip_infra_sync already has sufficient error handling, but for safety purpose adding another layer of try/catch can be safer

Reply:
Yeap. Maybe we can move some of the error handling stuff here so that we can remove some redundant ones from method below.

Reply:
One refactoring that I can think of that moves everything outside would be to catch one type of most probable exception outside and provide a useful logging message, such as stack does not exist exception. However the only disadvantage of such would be that tracing down other uncommon exceptions would be a bit harder as we will only print out the exact exception message without additional information